core/core_test.go

```diff
@@ -1175,3 +1184,219 @@ func TestValidateName(t *testing.T) {
 		t.Fatalf("No error expected when username and password provided, %v", err)
 	}
 }
+
+// Create a vault for the restore tests.
+func restoreCreateVault(t *testing.T, pstore, file string) {
+	cfg := config.New()
+	cfg.Delegations = &config.Delegations{
+		Persist:   true,
+		Mechanism: persist.FileMechanism,
+		Policy:    "Alice & Carl",
+		Users:     []string{"Alice", "Carl"},
+		Location:  pstore,
+	}
+
+	if err := Init(file, cfg); err != nil {
+		t.Fatalf("Unable to initialise core: %s", err)
+	}
+}
+
+func restoreState(t *testing.T, expected string) {
+	statusJson := []byte("{\"Name\":\"Admin\",\"Password\":\"Admin\"}")
+
+	var rd ResponseData
+	var st StatusData
+
+	if respJson, err := Status(statusJson); err != nil {
+		t.Fatalf("Failed to look up vault state: %s", err)
+	} else if err = json.Unmarshal(respJson, &rd); err != nil {
+		t.Fatalf("Error getting status, %v", err)
+	} else if err = json.Unmarshal(rd.Response, &st); err != nil {
+		t.Fatalf("Error getting status, %v", err)
+	} else if st.Status != expected {
+		t.Fatalf("Persistent delegations should be '%s' but are '%s'",
+			expected, st.Status)
+	}
+}
+
+// restoreBeforeRestart contains the test sequence to run before
```

Comment:
can we call them "beforeRestartRestore", and the other "afterRestartRestore" because restoreBeforeRestart sounds like a restore operation before restart.